src/main/java/com/o19s/es/ltr/logging/LoggingFetchSubPhase.java

```diff
@@ -221,6 +224,23 @@ public void accept(int featureOrdinal, float score) {
             currentLog.get(featureOrdinal).put("value", score);
         }
 
+        /**
+         * Return Map to store additional logging information returned with the feature values.
+         *
+         * The Map is created on first access.
+         */
+        @Override
+        public Map<String,Object> getExtraLoggingMap() {
+            if (extraLogging == null) {
+                extraLogging = new HashMap<>();
+                Map<String,Object> logEntry = new HashMap<>(2);
```

Comment:
nit: not sure it's worthwhile to start with 2 we might rehash since we add two 2 elements? I'd rather use the default initial capacity